dcpy/extract/metadata.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+import jinja2
+from datetime import datetime
+from pydantic import BaseModel
+import yaml
+
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import recipes, publishing
+from dcpy.library.validator import LibraryConfig
+from . import TEMPLATE_DIR, utils
+
+
+class Template(
+    BaseModel, use_enum_values=True, extra="forbid", arbitrary_types_allowed=True
+):
+    """Definition of a dataset for ingestion/processing/archiving in edm-recipes"""
+
+    name: str
+    acl: recipes.ValidAclValues
+
+    ## these two fields might merge to "source" or something equivalent at some point
+    ## for now, they are distinct so that they can be worked on separately
+    ## when implemented, "None" should not be valid type
+    source: recipes.ExtractConfig.Source.Options
```

Comment:
Could `Source.Options` live outside of `ExtractConfig`? Or is there a reason why it's packed under `ExtractConfig`? It feels confusing referencing `ExtractConfig` in `Template` because they seem to be independent entities...

Reply:
It was more thinking of the `ExtractConfig` as the "complete" object, but I think you're right. Ties in a bit to some of Alex's points around where these types live too. I think I wanted to not have too complex a web of interacting entities, less spread out, but per both your points its resulted in some types being in places that don't really match their domain

Reply:
Got it, thank you for clarification!